yogstation/code/modules/jobs/job_types/_job.dm

```diff
@@ -122,3 +122,35 @@
 	for(var/obj/effect/landmark/stationroom/box/bar/B in GLOB.landmarks_list)
 		template.load(B.loc, centered = FALSE)
 		qdel(B)
+
+/datum/job/proc/give_lawset_choice(mob/living/H, mob/M)
+
+	var/choice
+
+	var/client/C = M.client
+	var/mob/living/silicon/S = M
+	if (!C)
+		C = H.client
+		if (!C)
+			choice = "Random"
+	
+	if (!S)
```

Comment:
if there's no S you just return because something has gone horribly wrong.

Reply:
should I also return if there'd no C?

Reply:
as far as i can tell it doesn't matter with C, as long as you set it to Random when you've checked both